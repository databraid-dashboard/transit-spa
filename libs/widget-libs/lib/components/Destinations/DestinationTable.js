var _jsxFileName = '../../src/components/Destinations/DestinationTable.jsx',
    _this = this;

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Segment } from 'semantic-ui-react';
import DestinationRow from './DestinationRow';

export var DestinationTable = function DestinationTable(_ref) {
  var destinationIds = _ref.destinationIds,
      destinationsById = _ref.destinationsById;

  if (!destinationIds) return React.createElement(
    'div',
    {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 9
      },
      __self: _this
    },
    'Loading...'
  );

  return React.createElement(
    'div',
    {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 12
      },
      __self: _this
    },
    destinationIds.map(function (id) {
      return React.createElement(
        Segment,
        { key: id, __source: {
            fileName: _jsxFileName,
            lineNumber: 14
          },
          __self: _this
        },
        React.createElement(DestinationRow, { id: id, name: destinationsById[id].address, __source: {
            fileName: _jsxFileName,
            lineNumber: 15
          },
          __self: _this
        })
      );
    })
  );
};

DestinationTable.propTypes = {
  destinationIds: PropTypes.arrayOf(PropTypes.number).isRequired,
  destinationsById: PropTypes.shape({
    1: PropTypes.object
  }).isRequired
};

DestinationTable.defaultProps = {
  destinationIds: [1],
  destinationsById: { 1: {} }
};

export var mapStateToProps = function mapStateToProps(state) {
  var destinationIds = state.destinations.ids;
  var destinationsById = state.destinations.byId;

  return {
    destinationIds: destinationIds,
    destinationsById: destinationsById
  };
};

export default connect(mapStateToProps, null)(DestinationTable);