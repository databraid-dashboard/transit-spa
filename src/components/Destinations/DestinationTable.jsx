import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Segment } from 'semantic-ui-react';
import DestinationRow from './DestinationRow';

export const DestinationTable = ({ destinationIds, destinationsById }) => {
  if (!destinationIds) return <div>Loading...</div>;

  return (
    <div>
      {destinationIds.map(id =>
        (<Segment key={id}>
          <DestinationRow id={id} name={destinationsById[id].address} />
        </Segment>),
      )}
    </div>
  );
};

DestinationTable.propTypes = {
  destinationIds: PropTypes.arrayOf(PropTypes.number).isRequired,
  destinationsById: PropTypes.shape({
    1: PropTypes.object,
  }).isRequired,
};

DestinationTable.defaultProps = {
  destinationIds: [1],
  destinationsById: { 1: {} },
};

export const mapStateToProps = (state) => {
  const destinationIds = state.destinations.ids;
  const destinationsById = state.destinations.byId;

  return {
    destinationIds,
    destinationsById,
  };
};

export default connect(mapStateToProps, null)(DestinationTable);
