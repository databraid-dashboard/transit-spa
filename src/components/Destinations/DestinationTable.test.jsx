import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DestinationTable, mapStateToProps } from './DestinationTable';

describe('DestinationTable', () => {
  it('should render an empty table', () => {
    const component = shallow(<DestinationTable />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('mapStateToProps', () => {
    const component = shallow(<DestinationTable />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('mapStateToProps returns destinationIds and destinationsById', () => {
    const state = {
      widgets: {
        ids: ['transit'],
        byId: {
          transit: {
            configuration: {
              geolocating: false,
              currentLocation: {
                address: '44 Tehama St, San Francisco, CA 94105',
              },
            },
            destinations: {
              ids: [1],
              byId: {
                1: {
                  id: 1,
                  address: 'SFO, San Francisco, CA 94128',
                },
              },
            },
            journeys: {
              byDestinationId: {},
            },
          },
        },
      },
    };

    const expected = {
      destinationIds: [1],
      destinationsById: {
        1: {
          address: 'SFO, San Francisco, CA 94128',
          id: 1,
        },
      },
    };
    expect(mapStateToProps(state, { id: 1, widgetId: 'transit' })).toEqual(expected);
  });
});
