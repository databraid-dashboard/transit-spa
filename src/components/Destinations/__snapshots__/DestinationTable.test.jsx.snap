// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationTable should render an empty table 1`] = `
<div>
  <Component
    className="destination-row"
    id={1}
    name=""
    style={
      Object {
        "margin": "1rem 0rem",
      }
    }
  />
</div>
`;
